hs-bindgen/src/HsBindgen/C/Parser.hs

```diff
@@ -74,6 +79,35 @@ withTranslationUnit relPath tracer args fp k = do
         , CXTranslationUnit_VisitImplicitAttributes
         ]
 
+-- | Wrapper around 'clang_parseTranslationUnit', post-processing errors
+clang_parseTranslationUnit' ::
+     CXIndex
+  -> FilePath
+  -> ClangArgs
+  -> BitfieldEnum CXTranslationUnit_Flags
+  -> IO CXTranslationUnit
+clang_parseTranslationUnit' idx src args options =
+    handle callFailed $ clang_parseTranslationUnit idx src args options
+  where
+    -- If a call failed, it /might/ be because the file does exist (@libclang@
+    -- doesn't give us much info). Since this is the most likely reason, we
+    -- check for this here, throw a "does not exist" exception if the file does
+    -- not exist, and rethrow the original exception otherwise. Technically
+    -- speaking this is not correct: the file might have existed when we called
+    -- libclang, got removed /after/, and the 'CallFailed' exception was
+    -- unrelated. For now we simply accept this shortcoming.
```

Comment:
Add a sentence that there are other expected but probably rarer conditions, like permissions - file exists, but is unreadable etc.

I think it's ok to not be precice then, though i'm 99% sure the `CXError_ASTReadError` is returned for all those "IO" related issues.

Reply:
What about trying to open the file ourselves, and then just closing again, and relying on that for checking for existence, permissions, etc?

Reply:
Meh, this results in a confusing exception. I'll use the `2` API and check for some specific cases in postprocessing. Bear with me.